examples/init-function/init_function.bal

```diff
@@ -0,0 +1,16 @@
+// Uninitialized integer variable `value`.
+int value;
+
+// Uninitialized final string variable `name`.
+final string name;
+
+function init() returns error? {
+    // Initialize the variable `value` to 5.
+    value = 5;
+    // Initialize the final variable greeting to `James`.
+    name = "James";
+    
+    if value > 3 {
+        return error(string `${name}: The value should be less than 3, but it set to ${value}`);
+    }
+}
```

Comment:
Add a main function with a `io:println()` and explain it won't execute, because the  `init` function returned with an error.

Reply:
Updated 4d4430791d8cfa1d2ab3f39ab24fbfe5202f325b